src/odemis/driver/simstreakcam.py

```diff
@@ -0,0 +1,544 @@
+        self.binning = model.VAEnumerated(self._binning, choices_bin, setter=self._setBinning)
+        self._metadata[model.MD_BINNING] = self.binning.value
+
+        # physical pixel size is 6.5um x 6.5um
+        sensor_pixelsize = (6.5e-06, 6.5e-06)
+        self._metadata[model.MD_SENSOR_PIXEL_SIZE] = sensor_pixelsize
+
+        # pixelsize VA is the sensor size, it does not include binning or magnification
+        self.pixelSize = model.VigilantAttribute(sensor_pixelsize, unit="m", readonly=True)
+
+        # multiply with mag as we use the 1/M as input in yaml file!
+        eff_pixelsize = sensor_pixelsize[0] * self._binning[0] * self._metadata.get(model.MD_LENS_MAG, 1.0)
+        # self._metadata[model.MD_RESOLUTION] = eff_pixelsize  # TODO think it is useful
+
+        # Note: no function to get current acqMode.
+        # Note: Acquisition mode, needs to be before exposureTime!
+        # Acquisition mode should be either "Live" (non-sync acq) or "SingleLive" (sync acq) for now.
+        self.acqMode = "Live"
+
+        range_exp = [0.00001, 10]  # 10us to 10s
+        self._exp_time = 0.1  # 100 msec
+        self.exposureTime = model.FloatContinuous(self._exp_time, range_exp, unit="s", setter=self._setCamExpTime)
+        self._metadata[model.MD_EXP_TIME] = self.exposureTime.value
+
+        self.readoutRate = model.VigilantAttribute(425000000, unit="Hz", readonly=True)  # MHz
+        self._metadata[model.MD_READOUT_TIME] = 1 / self.readoutRate.value  # s
+
+        # spectrograph VAs after readout camera VAs
+        if self._spectrograph:
+            logging.debug("Starting streak camera with spectrograph.")
+            self._spectrograph.position.subscribe(self._updateWavelengthList, init=True)
+
+        # for synchronized acquisition
+        self._sync_event = None
+        self.softwareTrigger = model.Event()
+
+        # Simple implementation of the flow: we keep generating images and if
+        # there are subscribers, they'll receive it.
+        self.data = SimpleStreakCameraDataFlow(self._start, self._stop, self._sync)
+        self._generator = None
+
+    def _updateWavelengthList(self, _=None):
+        npixels = self.resolution.value[0]  # number of pixels, horizontal is wavelength
+        # pixelsize VA is sensor px size without binning and magnification
+        pxs = self.pixelSize.value[0] * self.binning.value[0] * self._metadata.get(model.MD_LENS_MAG, 1.0)
+        wll = self._spectrograph.getPixelToWavelength(npixels, pxs)
+        self._metadata[model.MD_WL_LIST] = wll
+
+    def _setBinning(self, value):
+        """
+        value (2-tuple int)
+        Called when "binning" VA is modified. It actually modifies the camera binning.
+        """
+        prev_binning, self._binning = self._binning, value
+
+        # adapt resolution
+        # TODO check if really necessary: why not just call resolutionFitter
+        # TODO self._binning is already updated and shape does not change
```

Comment:
What is this second TODO about? Indeed _binning is already updated and .shape is the maximum resolution, so it's fixed.